lib/player.rb

```diff
@@ -1,17 +1,35 @@
 module Scrabble
   class Player
-    attr_reader :name, :plays
+    attr_reader :name, :plays, :valid_word
+    attr_accessor :tiles
 
     def initialize(name)
       @name = name
       @plays = []
+      @tiles = []
+    end
+
+    def draw_tiles(tile_bag)
+      number = 7 - @tiles.length
+      @tiles += tile_bag.draw_tiles(number)
     end
 
     def play(word)
+      word = word.upcase
       if won?
         return false
       else
-        @plays.push(word)
+        word_in_array = word.split("")
+        remaining_tiles = @tiles.dup
+
+        word_in_array.each do |l|
+          remaining_tiles.slice!(remaining_tiles.index(l)) if remaining_tiles.include?(l)
```

Comment:
Awesome explanation, thanks!

While this is entirely a matter of taste and your understanding of intended audience... lines like this that are fairly "bulky" / "chunky" / full of code with very particular behavior are excellent candidates for having a comment in your code.

In this particular case what I would likely comment is:
```ruby
# This removes the first element in remaining_tiles that matches the letter l
```
in case anyone was unfamiliar with the `.index` or `.slice!` methods.